File: 2178/2178.c
int maximumSplitOfPositiveEvenInte(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximumSplitOfPositiveEvenInte(root->left);
    int r = maximumSplitOfPositiveEvenInte(root->right);
    return 1 + (l > r ? l : r);
}

File: 2178/2178.cs
public class Solution {
    public int MaximumSplitOfPositiveEvenInte(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximumSplitOfPositiveEvenInte(root.left), MaximumSplitOfPositiveEvenInte(root.right));
    }
}